jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigClassDocumentConfigurationSource.java

```diff
@@ -124,6 +132,36 @@ protected Object getBaseProperty(String propertyName, boolean text) throws XWiki
         return jiraServers;
     }
 
+    private JIRAServer getJiraServer(String authType, StringProperty urlProperty, StringProperty idProperty)
+    {
+        ComponentManager componentManager = componentManagerProvider.get();
+
+        JIRAServer jiraServer;
+        if (StringUtils.isEmpty(authType) || "noAuth".equals(authType)) {
+            jiraServer = new JIRAServer(urlProperty.getValue(), idProperty.getValue());
+        } else {
+            if (!componentManager.hasComponent(JIRAuthenticatorFactory.class, authType)) {
+                logger.error("Can't find JIRAuthenticatorFactory with name [{}]", authType);
+                jiraServer = new JIRAServer(urlProperty.getValue(), idProperty.getValue());
+            } else {
+                try {
+                    JIRAuthenticatorFactory jirAuthenticatorFactory = componentManagerProvider.get()
+                        .getInstance(JIRAuthenticatorFactory.class, authType);
+                    jiraServer = new JIRAServer(urlProperty.getValue(), idProperty.getValue(),
+                        jirAuthenticatorFactory.get(idProperty.getValue()));
+                } catch (ComponentLookupException e) {
+                    logger.error("Can't create JIRAuthenticatorFactory for authentication type [{}]", authType,
+                        e);
+                    return null;
```

Comment:
would be cleaner to initialize jiraServer at null so you'd always return null in the last statement in case of problem and you wouldn't have to do anything else than logging an error if the hint is not found in CM